Hookies/Contexts/GamePlay/GameScene.swift

```diff
@@ -77,20 +79,23 @@ class GameScene: SKScene, SKPhysicsContactDelegate {
      // MARK: - Collision Detection
 
     func didBegin(_ contact: SKPhysicsContact) {
-        guard has(contact: contact, with: currentPlayer) else {
-                return
-        }
-
-        if has(contact: contact, with: finishingLine) {
-            gameEngine?.stopCurrentPlayer()
-        }
-
-        for powerup in powerups where has(contact: contact, with: powerup) {
-            handleContactWithPowerup(powerup)
-        }
-
-        for trap in traps where has(contact: contact, with: trap) {
-            handleContactWithTrap(trap)
+        for player in localPlayers {
+            if has(contact: contact, with: player) {
+                if has(contact: contact, with: finishingLine) {
+                    gameEngine?.stopLocalPlayer(playerNode: player)
+                    return
+                }
+                for trap in traps where has(contact: contact, with: trap) {
+                    handleContactWithTrap(between: player, trap: trap)
+                    return
+                }
+                if player == currentPlayer {
```

Comment:
Would bot be able to collect powerups?

Reply:
No, only currentPlayer can collect powerups